sound/soc/sof/sof-pci-dev.c

```diff
@@ -150,6 +150,24 @@ static const struct dev_pm_ops sof_pci_pm = {
 			   NULL)
 };
 
+static void sof_pci_probe_complete(struct device *dev)
+{
+	dev_dbg(dev, "Completing SOF PCI probe");
+
+	/* allow runtime_pm */
+	pm_runtime_set_autosuspend_delay(dev, SND_SOF_SUSPEND_DELAY_MS);
+	pm_runtime_use_autosuspend(dev);
+
+	/*
+	 * runtime pm for pci device is "forbidden" by default.
+	 * so call pm_runtime_allow() to enable it.
+	 */
+	pm_runtime_allow(dev);
```

Comment:
@plbossart I am wondering what would happen if the dsp probe failed in sof_probe() failed and we never enabled PM?
What would be the impact of calling pm_runtime_get_noresume() in sof_pci_remove()?

Reply:
@ranj063 you tell me :-)
I tested this without any firmware files, and didn't see any impact. I could add/remove modules, fail the probe and loop forever.